vault/provider.go

```diff
@@ -282,6 +282,14 @@ var (
 			Resource:      genericSecretDataSource(),
 			PathInventory: []string{"/secret/data/{path}"},
 		},
+		"vault_generic_secret_list": {
+			Resource:      genericSecretListDataSource(),
+			PathInventory: []string{"/secret/data/{path}/?list=true"},
+		},
+		"vault_generic_secret_list_v2": {
```

Comment:
Now I'm wondering if we should just name the datasource `vault_kv_secret_list` and `vault_kv_secret_list_v2` ?  We could document the relationship to `vault_generic_secret`